### ISSUE TITLE
CyberRT topics to ROS and vice versa

### ISSUE THREAD

**Issue by Prj1508** [ApolloAuto/apollo#13374]
I am working on an external component that has ROS interface. I would like to connect Apollo cyber topics into ROS topics. Is there a way to convert CyberRT topics and messages into ROS topics/messages?

**Comment by SeasoulChris** [comment]
maybe you can try apollo_ros_bridge

**Comment by lx18233184051** [comment]
> maybe you can try apollo_ros_bridge

apollo_ros_bridge, this project is on github. but It's not offical. maybe you can try.

**Comment by Thinsheep** [comment]
Hi @Prj1508,
Have you tried if apollo_ros_bridge is working successfully or have you found another solution?

@lx18233184051 and @SeasoulChris 
I am planning to use the other simulator for my project ( VIRES VTD) not LGSVL.  As far as I know, there is no
Cyber RT integration available yet. VIRES VTD offers ROS integration and API for C ++ Integration.
There are bridges connection between LGSVL to Apollo and Carla to Apollo. Is there any chance to
modify these bridges to use them for VIRES VTD to Apollo? If you have knowledge in this area please give
me some suggestions. I am totally new to this area and just start my project. 
Thank you.

**Comment by Prj1508** [comment]
Yes, the bridge is working fine. If your simulator has ROS interface, then Apollo+  apollo_ros_bridge should do it for you I think. Any specific reason you are not using LGSVL or Carla? Just curious

**Comment by Thinsheep** [comment]
Hi @Prj1508 
Thanks for your reply, actually, we already have LGSVL, we just want to try a commercial simulator for the comparison. We are developing the simulation testing technique.

**Comment by Thinsheep** [comment]
Thank you.
It is very nice. Have you installed Apollo 6.0 on your PC? I am currently looking for a PC to install Apollo 6.0 and Simulator. I am going to go with GPU RTX 3090/3080 but I have read the information that "NVIDIA Turing GPU (2080/2080 Ti) is strongly recommended". Do you know if RTX 3090/3080 (faster and not much more expensive but not Turing GPU) is ok to run Apollo 6.0?

**Comment by Thinsheep** [comment]
Hi Prj1508,
I need Vires VTD communicating with Apollo 6.0. I need to build a ROS interface for Vires VTD as they only have an example for 1 signal not completed package (I do not know if Vires VTD was developed for Autonomous software platform because they do not build the communication bridge with current popular AV stacks). I have a few questions below if you have experience with this please share some information.

1. To build the interface for Vires VTD, I need to know which message VTD will need to provide-publish to and which message VTD can get-subscribe from Apollo. VTD will send sensor data and get the AV commands (accelerator, brake, throttle). How can I get the full list of sensor data and AV commands so I do not miss any interface?

2. Is apollo_ros_bridge is the completed package that I can directly use for Apollo 6.0? because they have only some ROS topics like Path, Odometry, PointCloud2 and IMU but not all. I also do not see if they publish AV commands like accelerator, brake and throttle.

3. Can Vires VTD interface and apollo_ros_bridge can publish/subcribe data directly or do I have to build something to establish the communication?

**Comment by Thinsheep** [comment]
Hi @Prj1508 ,
I am looking for a tutor for this topic. If you are available please drop me an email at sgttrieu@gmal.com, we can discuss further :).

**Comment by AlexandrZabolotny** [comment]
> Yes, the bridge is working fine. If your simulator has ROS interface, then Apollo+ apollo_ros_bridge should do it for you I think. Any specific reason you are not using LGSVL or Carla? Just curious
> […](#)

Hi @Prj1508 

I will be glad if you can help me!  

I'm trying to figure it out with apollo_ros_bridge.  
https://github.com/AuroAi/apollo_ros_bridge  


But when I run the bridge in docker, then the frequency of received channels in apollo_ros_bridge does not match the frequency of messages in Apollo.  
I am attaching a photo showing the frequencies.  
Maybe you have an idea how to fix this ??  
![Screenshot from 2021-03-30 09-48-24](https://user-images.githubusercontent.com/61351599/112954964-985efe80-9147-11eb-81e9-625c1342bf11.png)

**Comment by Thinsheep** [comment]
Hi @AlexandrZabolotny 
apollo_ros_bridge was developed for Apollo 3.5. Do you know any chance to use this apollo_ros_bridge for Apollo 6.0?

**Comment by AlexandrZabolotny** [comment]
> Hi @AlexandrZabolotny
> apollo_ros_bridge was developed for Apollo 3.5. Do you know any chance to use this apollo_ros_bridge for Apollo 6.0?

bridge also works in version 6

**Comment by Thinsheep** [comment]
> > Hi @AlexandrZabolotny
> > apollo_ros_bridge was developed for Apollo 3.5. Do you know any chance to use this apollo_ros_bridge for Apollo 6.0?
> 
> bridge also works in version 6

Thanks for your information.
Importing these three packages to Apollo 6.0 Master is the right idea to run the bridge in version 6.0?
![image](https://user-images.githubusercontent.com/52489127/114457031-da735000-9bdd-11eb-9e8f-91a0d5474324.png)

**Comment by AlexandrZabolotny** [comment]
> > > Hi @AlexandrZabolotny
> > > apollo_ros_bridge was developed for Apollo 3.5. Do you know any chance to use this apollo_ros_bridge for Apollo 6.0?
> > 
> > 
> > bridge also works in version 6
> 
> Thanks for your information.
> Importing these three packages to Apollo 6.0 Master is the right idea to run the bridge in version 6.0?
> ![image](https://user-images.githubusercontent.com/52489127/114457031-da735000-9bdd-11eb-9e8f-91a0d5474324.png)

I use Apollo pre6

**Comment by Thinsheep** [comment]
@AlexandrZabolotny 
Sorry, I am lacking knowledge in this. This bridge was built in Apollo 3.5. How did you transfer it into pre6?

**Comment by Shadow-Haund** [comment]
> > Hi @AlexandrZabolotny
> > apollo_ros_bridge was developed for Apollo 3.5. Do you know any chance to use this apollo_ros_bridge for Apollo 6.0?
> 
> bridge also works in version 6

@AlexandrZabolotny, hello.
Maybe you know, does apollo_ros_bridge work with Apollo ver 8.0?
